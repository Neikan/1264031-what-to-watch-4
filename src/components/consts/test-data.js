export const Promo = {
  TITLE: `The Grand Budapest Hotel poster`,
  GENRE: `Drama`,
  DATE: 2014
};

export const MOVIES = [
  {
    id: `mock-1`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: `fantastic-beasts-the-crimes-of-grindelwald`
  }, {
    id: `mock-2`,
    title: `Bohemian Rhapsody`,
    poster: `bohemian-rhapsody`
  }, {
    id: `mock-3`,
    title: `Macbeth`,
    poster: `macbeth`
  }
];

