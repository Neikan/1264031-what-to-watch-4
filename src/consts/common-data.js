export const ALL_GENRES = `All genres`;


export const AuthStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};


export const BASE_URL = `https://4.react.pages.academy`;


export const COUNT_GENRES = 10;


export const CountMovies = {
  ALL: 20,
  START: 8,
  LIKED_BY_GENRE: 4
};


export const DECIMAL_BASE = 10;


export const FormatRule = {
  REVIEW_DATE: `MMMM DD, YYYY`,
  REVIEW_DATE_HIDE: `YYYY-MM-DD`
};


export const HIDDEN_CLASS = `visually-hidden`;


export const LogoPosition = {
  HEADER: `header`,
  FOOTER: `footer`
};


export const MovieTabList = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`,
};


export const PAD_STRING_ZERO = `0`;


export const Page = {
  MAIN: `/`,
  MOVIE: `films`,
  LOGIN: `login`,
  PLAYER: `player`,
  ADD_REVIEW: `review`
};


export const Rating = {
  DEFAULT: 3,
  WEIGHT: 2
};


export const RATING_STARS = [1, 2, 3, 4, 5];


export const ReviewMessage = {
  DEFAULT: ``,
  LENGTH_MIN: 50,
  LENGTH_MAX: 400
};


export const ReviewMessageStatus = {
  DEFAULT: ``,
  SUCCESS: `Ваш комментарий добавлен!`,
  ERROR_SENDING: `Извините, при добавлении Вашего комментария возникла ошибка. Попробуйте позднее.`,
  MESSAGE_LENGTH: `Текущая длина сообщения:`
};


export const REVIEW_PLACEHOLDER = `Введите сообщение не менее ${ReviewMessage.LENGTH_MIN} и не более ${ReviewMessage.LENGTH_MAX} символов`;


export const RequestStatusCode = {
  OK: 200,
  UNAUTHORIZED: 401
};


export const TextualRating = {
  BAD: {
    level: `Bad`,
    from: 0,
  },
  NORMAL: {
    level: `Normal`,
    from: 3
  },
  GOOD: {
    level: `Good`,
    from: 5
  },
  VERY_GOOD: {
    level: `Very good`,
    from: 8
  },
  AWESOME: {
    level: `Awesome`,
    from: 10
  }
};


export const Time = {
  MINUTES: 60,
  SECONDS: 60
};


export const Timeout = {
  WAIT: 2000,
  REQUEST: 5000
};


export const Url = {
  WTW: `wtw`,
  FILMS: `films`,
  COMMENTS: `comments`,
  PROMO: `promo`,
  LOGIN: `login`
};
