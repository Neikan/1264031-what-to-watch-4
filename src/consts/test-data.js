export const LogoPosition = {
  HEADER: `header`,
  FOOTER: `footer`
};


export const Promo = {
  TITLE: `The Grand Budapest Hotel poster`,
  GENRE: `Drama`,
  DATE: 2014
};


export const MOVIES = [
  {
    id: `mock-1`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: `fantastic-beasts-the-crimes-of-grindelwald`
  }, {
    id: `mock-2`,
    title: `Bohemian Rhapsody`,
    poster: `bohemian-rhapsody`
  }, {
    id: `mock-3`,
    title: `Macbeth`,
    poster: `macbeth`
  }
];


export const GENRES = [
  {
    id: `g1`,
    title: `All genres`,
  }, {
    id: `g2`,
    title: `Comedies`,
  }, {
    id: `g3`,
    title: `Crime`
  }
];


export const MarkupElement = {
  PROMO_TITLE: `movie-card__title`,
  MOVIE_CARD_TITLE: `small-movie-card__title`,
  GENRE_LINK: `catalog__genres-link`
};
