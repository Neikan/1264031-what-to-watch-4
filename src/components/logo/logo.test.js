import React from "react";
import renderer from "react-test-renderer";
import Logo from "./logo.jsx";
import {LogoPosition} from "./../../consts/test-data";


describe(`Test Logo component`, () => {
  test(`Logo component is created and rendered correctly for Header`, () => {
    const tree = renderer
      .create(
          <Logo logoPosition = {LogoPosition.HEADER}/>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });


  test(`Logo component is created and rendered correctly for Footer`, () => {
    const tree = renderer
      .create(
          <Logo logoPosition = {LogoPosition.FOOTER}/>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
