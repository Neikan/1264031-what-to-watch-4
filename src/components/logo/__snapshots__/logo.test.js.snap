// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Logo component Logo component is created and rendered correctly for Footer 1`] = `
<div
  className="logo"
>
  <a
    className="logo__link logo__link--light"
  >
    <span
      className="logo__letter logo__letter--1"
    >
      W
    </span>
    <span
      className="logo__letter logo__letter--2"
    >
      T
    </span>
    <span
      className="logo__letter logo__letter--3"
    >
      W
    </span>
  </a>
</div>
`;

exports[`Test Logo component Logo component is created and rendered correctly for Header 1`] = `
<div
  className="logo"
>
  <a
    className="logo__link"
  >
    <span
      className="logo__letter logo__letter--1"
    >
      W
    </span>
    <span
      className="logo__letter logo__letter--2"
    >
      T
    </span>
    <span
      className="logo__letter logo__letter--3"
    >
      W
    </span>
  </a>
</div>
`;
